src/game_events/manager.cpp

```diff
@@ -190,24 +227,8 @@ void manager::execute_on_events(const std::string& event_id, manager::event_func
 			}
 
 			// Could be more than one.
-			for (const std::string& name : handler->names()) {
-				bool matches = false;
-
-				if (utils::might_contain_variables(name)) {
-					// If we don't have gamedata, we can't interpolate variables, so there's
-					// no way the name will match. Move on to the next one in that case.
-					if (!gd) {
-						continue;
-					}
-
-					matches = standardized_event_id ==
-						event_handlers::standardize_name(utils::interpolate_variables_into_string(name, *gd));
-				}
-				else {
-					matches = standardized_event_id == name;
-				}
-
-				if (matches) {
+			for(const std::string& name : handler->names(gd)) {
```

Comment:
The old code checked for `gd != nullptr`, the new code looks as if it will unconditionally try to dereference. What guards against that?

Reply:
"some places" is an extremely unhelpful comment.

Reply:
Where those "some places" are depends on how the unconditional dereference is fixed. Either make `handler->names` handle a potential nullptr, or make it take a reference instead.

Reply:
Regardless, I addressed this in 28d555c.

Reply:
So, was the check for `utils::might_contain_variables(name)` just dead code? I mean, there was a code path specifically for events that occurred when `gd` was `nullptr`, including a specific path which would allow ones that didn't need interpolation to still happen.

Reply:
???

What are you even talking about? That check is still there in `handler->names`, protected by a null check on the passed-in `variable_set`.

Reply:
I missed the check inside `handler->names`, there wasn't an unconditional dereference, sorry about that part.

28d555c appears to be incorrect, and it's 28d555c that caused the "dead code" question. It means we can't reach the `utils::might_contain_variables(name)` line with `gd` being `nullptr`, which means the moved-but-still-present conditional block for `gd == nullptr && !utils::might_contain_variables(single_name)` is dead code. The question about whether that conditional block is actually useful remains.

Also, in the new code (but not the old) the check for `utils::might_contain_variables(single_name)` is being done after the variables have been interpolated.

Reply:
> Also, in the new code (but not the old) the check for `utils::might_contain_variables(single_name)` is being done after the variables have been interpolated.

???

It's literally "if might contain variables then interpolate variables"...

Reply:
Okay, it took me a bit to figure out what you were saying, but… basically, the first check is wrong. It should split, _then_ interpolate, right?

Reply:
I'm assuming that's been done, at least it looks done from a quick check.